### PR TITLE
Added authorization prompt in Mitre > Intelligence

### LINKED ISSUE #4243: Inappropriate permissions' handling in MITRE intelligence section
Hello,

The RBAC permissions in the MITRE intelligence section of the Wazuh Dashboard are not being properly handled.

If we enter the MITRE inventory section without read permissions, the following errors appear:

image

### PR DESCRIPTION
**Pull request by Machi3mfl**
Hi Team,
This PR add the AuthorizationPrompt in Mitre Intelligence section

Closes #4243

### CODE REVIEW

File: public/components/overview/mitre_attack_intelligence/intelligence.tsx [hunk 1]
```diff
@@ -12,62 +12,77 @@
...
 import { ModuleMitreAttackIntelligenceLeftPanel } from './intelligence_left_panel';
 import { ModuleMitreAttackIntelligenceRightPanel } from './intelligence_right_panel';
 import { useAsyncAction } from '../../common/hooks';
 import { WzRequest } from '../../../react-services';
 import { PanelSplit } from '../../common/panels';
+import { withUserAuthorizationPrompt } from '../../common/hocs';
+import { compose } from 'redux';
 
-export const ModuleMitreAttackIntelligence = () => {
+const ModuleMitreAttackIntelligenceComponent = () => {
```
Comment (Desvelao) [inline public/components/overview/mitre_attack_intelligence/intelligence.tsx:26]: **suggestion**: we could define the component to use directly including the HOC and not create a intermediate component.

```tsx
export const ModuleMitreAttackIntelligence = compose(
  withUserAuthorizationPrompt([{ action: 'mitre:read', resource: '*:*:*' }])
)(() => {
  /* Component definition */
});
```

**Comment by Desvelao** [comment]
**review**:
test :heavy_check_mark: 

## Screenshot
image

File: public/components/overview/mitre_attack_intelligence/intelligence.tsx (hunk 1 above)
Comment (Machi3mfl) [reply public/components/overview/mitre_attack_intelligence/intelligence.tsx]: Removed intermediate component and added tests to the component composed with permissions hoc

**Review by Desvelao — APPROVED** [review]
**review**:
code :heavy_check_mark: 
test :heavy_check_mark: 

# Test suite
```
kibana@kibana_dev:/usr/share/kibana/plugins/wazuh$ yarn test:jest public/components/overview/mitre_attack_intelligence/intelligence.test.tsx
yarn run v1.21.1
$ node scripts/jest public/components/overview/mitre_attack_intelligence/intelligence.test.tsx
 PASS  public/components/overview/mitre_attack_intelligence/intelligence.test.tsx (9.643 s)
  Module Mitre Att&ck intelligence container
    ✓ should render the component if has permissions (58 ms)
    ✓ should render permissions prompt when no has permissions (7 ms)
```

## Screenshot
image

**Comment by chantal-kelm** [comment]
code :heavy_check_mark: 
test :heavy_check_mark: 

Screenshot from 2022-06-16 15-36-30

**Pull request merged**